pkg/skaffold/event/v2/event.go

```diff
@@ -190,6 +241,19 @@ func emptyStatusCheckState() *proto.StatusCheckState {
 	}
 }
 
+func AutoTriggerDiff(name string, val bool) (bool, error) {
+	switch name {
+	case "build":
```

Comment:
we could reuse `pkg/errors.Phase` type and add `sync` to it?

Reply:
Good point! I'll update this

Reply:
https://github.com/GoogleContainerTools/skaffold/pull/5625